src/game/Movement/FleeingMovementGenerator.cpp

```diff
@@ -61,6 +61,7 @@ void FleeingMovementGenerator<T>::_setTargetLocation(T &owner)
     Movement::MoveSplineInit init(owner, "FleeingMovementGenerator");
     init.Move(&path);
     init.SetWalk(_forceWalking);
+    _customSpeed = ((Creature*)&owner)->GetFleeingSpeed();
```

Comment:
FleeingMovementGenerator is not templated just for creatures. It should be using the speed of their current movement type in any case, so setting a custom velocity would be redundant.